rts/Sim/MoveTypes/GroundMoveType.cpp

```diff
@@ -512,6 +512,10 @@ CGroundMoveType::CGroundMoveType(CUnit* owner):
 
 	flatFrontDir = (owner->frontdir * XZVector).Normalize();
 
+	// Override the unit size, it should match the MoveDef's to avoid conflicts elsewhere in the code.
+	owner->xsize = md->zsize;
+	owner->zsize = md->xsize;
```

Comment:
Seems to set owner X to md Z and vice versa?

Reply:
It would be good if the unit def's footprint was also set to its default movedef's footprint. Probably don't do it here though because individual units can have a movedef reassigned dynamically to something different than their default, and that shouldn't cause other units of the same type to change their size.

Reply:
Good spot! I've fixed the order. I'll look at what it may take to sensibly update the unit def.

Reply:
I'm no longer sure if updating the unit def is a good idea. Maybe let's keep this part as-is and see if there are issues in practice?

Reply:
Sure.